File: leetcode/1easy/위성원/2765.ts
function alternatingSubarray(nums: number[]): number {
  let answer = -1;
  for (let i = 0; i < nums.length - 1; i++) {
    let mode = "minus"
    let curr = nums[i];
    let sub = [curr]
    for (let j = i + 1; j < nums.length; j++) {
      const next = nums[j];

      if (mode === "minus") {
        if (next - curr === 1) {
          curr = next;
          mode = "plus"
          sub.push(next)
        } else {
          break;
        }
      } else if (mode === "plus") {
        if (next - curr === -1) {
          curr = next;
          mode = "minus"
          sub.push(next)
        } else {
          break;
        }
      }
    }

    if (sub.length > 1) {
      answer = Math.max(answer, sub.length)
    }
  }

  return answer
};

File: leetcode/1easy/이경호/2765.cpp
class Solution {
public:
  int alternatingSubarray(vector<int>& nums) {
    int start = 0;
    int end = 1;
    int ans = -1;
    int cnt = 1;
    while (true) {
      if (end == nums.size()) {
        ans = std::max(ans, cnt);
        return ans > 1 ? ans : -1;
      }
      if ((nums[end] - nums[end - 1]) == (((end - start) % 2) ? 1 : -1)) {
        cnt++;
        end++;
        continue;
      }
      else {
        ans = std::max(ans, cnt);
        cnt = 1;
        if ((end - start) % 2) {
          start = end;
          end++;
          continue;
        }
        else {
          start = end - 1;
          continue;
        }
      }
    }
  }
};